telemetry-schema.md

```diff
@@ -0,0 +1,71 @@
+## Attributes
+
+Key | Description
+--- | ---
+event | `perf-report` (can this take other values?)
+results | String holding a JSON-encoded `OuterResults`
+sentCount | string-encoded integer representing the number of `Results` in each `Test`
+
+## OuterResults
+
+Key | Description
+--- | ---
+status | `"complete"` (Can this take other values?)
+tests | array of `Test`
+
+## Test
+
+Key | Description
+--- | ---
+ccheck | ?
+doh | `true` or `false`; whether DOH was used to resolve the remote endpoint
+facebook | `true` or missing; whether the remote endpoint is Facebook
+label | friendly slug with a 1:1 mapping to a (`doh`, `url`) pair
+results | array of `Result`
+url | URL of endpoint
+
+## Result
+
+Key | Description
+--- | ---
+asyncOpenTime | timestamp
+channelCreationTime | timestamp
+connectEndTime | timestamp
+connectStartTime | timestamp
+domainLookupEndTime | timestamp
+domainLookupStartTime | timestamp
+errorCode | `0` or missing?
+event | `"loadend"` (Can this take other values?)
+headers | array of `Header`
+isTRR | `true` or missing; whether DOH was used to resolve the remote endpoint
+remoteAddress | IPv[46] address of the fetch endpoint
+requestStartTime | timestamp
+responseCode | HTTP status code for the fetch
+responseEndTime | timestamp
+responseStartTime | timestamp
+secureConnectionStartTime | timestamp or `0`?
+securityState | `2` or missing?
```

Comment:
I think it can be any integer value: https://searchfox.org/mozilla-central/source/netwerk/socket/nsITransportSecurityInfo.idl#14
It will be missing in the case of an insecure connection but also sometimes it's not available in secure requests too like many of these attributes but I suspect we should treat those as errors.

Reply:
I thiiiiiink it's a bitfield and some of the components are defined at https://searchfox.org/mozilla-central/rev/3e0f1d95fcf8832413457e3bec802113bdd1f8e8/uriloader/base/nsIWebProgressListener.idl#153-175 -- that's consistent with "2" being a success, anyway